xonsh/completers/environment.py

```diff
@@ -27,4 +28,8 @@ def complete_environment_vars(context: CompletionContext):
     filter_func = get_filter_function()
     env_names = XSH.env
 
-    return {"$" + k for k in env_names if filter_func(k, key)}, lprefix
+    return {
+        RichCompletion("$" + k, description=env_names.get_docs(k).doc)
+        for k in env_names
+        if filter_func(k, key)
+    }, lprefix
```

Comment:
perhaps a generator. we had some discussion on ordering

Reply:
Done